nilearn/load_confounds/parser.py

```diff
@@ -0,0 +1,387 @@
+]
+
+
+def _sanitize_strategy(strategy):
+    """Defines the supported denoising strategies."""
+    if not isinstance(strategy, list):
+        raise ValueError("strategy needs to be a list of strings")
+    for conf in strategy:
+        if conf not in all_confounds:
+            raise ValueError(f"{conf} is not a supported type of confounds.")
+    # add non steady state if not present
+    if "non_steady_state" not in strategy:
+        strategy.append("non_steady_state")
+    return strategy
+
+
+def _check_error(missing_confounds, missing_keys):
+    """Consolidate a single error message across multiple missing confounds."""
+    if missing_confounds or missing_keys:
+        error_msg = (
+            "The following keys or parameters are missing: "
+            + f" {missing_confounds}"
+            + f" {missing_keys}"
+            + ". You may want to try a different denoising strategy."
+        )
+        raise ValueError(error_msg)
+
+
+class Confounds:
+    """
+    Confounds from fmriprep
+
+    To enable easy confound variables loading from fMRIprep outputs, Confounds
+    provides a interface that groups subsets of confound variables into noise
+    components and their parameters. It is possible to fine-tune a subset of
+    noise components and their parameters through Confounds.
+
+    The implementation will only support fMRIPrep output from the 1.2.x series.
+    The `compcor` noise component requires 1.4.x series or above.
+
+    Parameters
+    ----------
+    strategy : list of strings
+        The type of noise components to include.
+        "motion" head motion estimates.
+        "high_pass" discrete cosines covering low frequencies.
+        "wm_csf" confounds derived from white matter and cerebrospinal fluid.
+        "global" confounds derived from the global signal.
+        "ica_aroma" confounds derived from ICA-AROMA.
+        "scrub" regressors for Power 2014 scrubbing approach.
+
+    motion : string, optional
+        Type of confounds extracted from head motion estimates.
+        "basic" translation/rotation (6 parameters)
+        "power2" translation/rotation + quadratic terms (12 parameters)
+        "derivatives" translation/rotation + derivatives (12 parameters)
+        "full" translation/rotation + derivatives + quadratic terms + power2d
+               derivatives (24 parameters)
+
+    n_motion : float
+        Number of pca components to keep from head motion estimates.
+        If the parameters is strictly comprised between 0 and 1, a principal
+        component analysis is applied to the motion parameters, and the number
+        of extracted components is set to exceed `n_motion` percent of the
+        parameters variance.
+        If the n_components = 0, then no PCA is performed.
+
+    fd_thresh : float, optional
+        Framewise displacement threshold for scrub (default = 0.2 mm)
+
+    std_dvars_thresh : float, optional
+        Standardized DVARS threshold for scrub (default = 3)
```

Comment:
Sorry but what is DVARS ? (I don't like these accronyms...)

Reply:
I will keep my personal opinion about this acronym to myself :stuck_out_tongue: . Here's the explanation from [MRIQC docs](https://mriqc.readthedocs.io/en/stable/iqms/bold.html#measures-for-the-temporal-information):

> DVARS - D referring to temporal derivative of timecourses, VARS referring to RMS variance over voxels ([Power2012] dvars_nstd) indexes the rate of change of BOLD signal across the entire brain at each frame of data.

Reply:
Certainly we can expand this.